File: Chef.java
public class Chef {
    private RecetaBuilder recetaBuilder;

    public void setRecetaBuilder(RecetaBuilder rb) {
        recetaBuilder = rb;
    }

    public Receta getReceta() {
        return recetaBuilder.getReceta();
    }
    public void BuildReceta() {
        recetaBuilder.BuildIngredientes();
        recetaBuilder.BuildPasos();
        recetaBuilder.BuildTiempo();
    }
}

File: practica4/recetario_flutter/lib/chef.dart
import 'recetabuilder.dart';
import 'receta.dart';

class Chef {
    RecetaBuilder? recetaBuilder;

    void setRecetaBuilder(RecetaBuilder rb) {
      recetaBuilder = rb;
    }

    Receta getReceta() {
      if (recetaBuilder == null) {
        throw Exception('RecetaBuilder no ha sido construido');
      }
      return recetaBuilder!.getReceta();
    }

    void buildReceta() {
        if (recetaBuilder == null) {
            throw Exception('RecetaBuilder no ha sido construido');
        }
        recetaBuilder!.buildIngredientes();
        recetaBuilder!.buildPasos();
        recetaBuilder!.buildTiempo();
    }
}
